UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RnwRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using UKHO.MaritimeSafetyInformation.Common;
+using UKHO.MaritimeSafetyInformation.Common.Logging;
+using UKHO.MaritimeSafetyInformation.Common.Models.RadioNavigationalWarning.DTO;
+
+namespace UKHO.MaritimeSafetyInformation.Web.Services
+{
+    public class RnwRepository : IRnwRepository
+    {
+        private readonly RadioNavigationalWarningsContext _context;
+        private readonly ILogger<RnwRepository> _logger;
+
+        public RnwRepository(RadioNavigationalWarningsContext context,
+                             ILogger<RnwRepository> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+        public async Task<bool> AddRadioNavigationWarnings(RadioNavigationalWarnings radioNavigationalWarnings, string correlationId)
+        {
+            try
+            {
+                if (radioNavigationalWarnings.WarningType != 0 && radioNavigationalWarnings.Reference != "" && radioNavigationalWarnings.DateTimeGroup.HasValue 
+                    && radioNavigationalWarnings.Summary != null && radioNavigationalWarnings.Content != null)
```

Comment:
Move this business logic to service layer, as repository is responsible for dealing with database interactions (add/save etc.)